### PR TITLE
Full project restructure

### PR DESCRIPTION
**Pull request by nitreo**
We changed the way we export the API.
Consumer should now require the container: `var pf = require('playfab-sdk');`
After that different apis are accessible through the container:
`var adminApi = pf.PlayFabAdmin`.

Tests are now part of the main package and reside inside tests folder.
One can run tests by executing `npm test` from the cmd, while cwd is {repo}/PlayFabSdk

Prior to first running tests, you have to execute `npm install`. This will install nodeunit dependencies.
This does not eliminate requirement on nodeunit installed globally!

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,37 @@
+{
```
Comment (pgilmorepf) [inline .vscode/launch.json]: Please add .vscode folder to .gitignore

File: PlayFabSdk/tests/.gitignore [hunk 2]
```diff
@@ -0,0 +1 @@
+/testTitleData.js
```
Comment (pgilmorepf) [inline PlayFabSdk/tests/.gitignore]: testTitleData.json?

Comment (pgilmorepf) [reply PlayFabSdk/tests/.gitignore]: Also, this can probably be in the top level .gitignore...

File: PlayFabSdk/tests/run.bat [hunk 3]
```diff
@@ -0,0 +1,5 @@
+cd tests
+set cd=%cd%
+IF "%PF_TEST_TITLE_DATA_JSON%"=="" set PF_TEST_TITLE_DATA_JSON=./testTitleData.js
```
Comment (pgilmorepf) [inline PlayFabSdk/tests/run.bat]: testTitleData.js -> testTitleData.json

File: PlayFabSdk/tests/.gitignore (hunk 2 above)
Comment (nitreo) [reply PlayFabSdk/tests/.gitignore]: Yes, this allows you to store your testTitleData.json and this won't get propagated to any repositiory. 
By default, testTitleData.json gets read from ./tests folder. You can still specify PF_TEST_WHATEVER enveronmental variable like you did before. 
Ideally, we should add generator for the testTitleData if nothing is found.

**Comment by pgilmorepf** [comment]
This was ported directly into the SdkGenerator, and built into this repo from there:
https://github.com/PlayFab/SDKGenerator/commit/c364031114b93c214814eb95482055d0c699bbe8

**Pull request closed without merging**